layers/+lang/typescript/packages.el

```diff
@@ -98,23 +101,36 @@
                                   (goto-char p)
                                   (message "formatted.")
                                   (kill-buffer outputbuf))
-                                (progn 
-                                  (message "Formatting failed!")
-                                  (display-buffer outputbuf)))
+                              (progn
+                                (message "Formatting failed!")
+                                (display-buffer outputbuf)))
                             (progn
                               (delete-file tmpfile)))))
                   (message "tsfmt not found. Run \"npm install -g typescript-formatter\"")))
 
+              (defun typescript/open-region-in-playground (start end)
+                "Open selected region in http://www.typescriptlang.org/Playground
+                 If nothing is selected - open the whole current buffer."
+                (interactive (if (use-region-p)
+                                 (list (region-beginning) (region-end))
+                                 (list (point-min) (point-max))))
+                (browse-url (concat "http://www.typescriptlang.org/Playground#src="
+                                     (url-hexify-string (buffer-substring-no-properties start end)))))
+
               (spacemacs/set-leader-keys-for-major-mode 'typescript-mode "=" 'typescript/format-buffer)
 
               (when typescript-fmt-on-save
 
-                    (defun typescript/before-save-hook ()
-                      (add-hook 'before-save-hook 'typescript/format-buffer t t))
+                (defun typescript/before-save-hook ()
+                  (add-hook 'before-save-hook 'typescript/format-buffer t t))
 
-                    (add-hook 'typescript-mode-hook 'typescript/before-save-hook)))))
+                (add-hook 'typescript-mode-hook 'typescript/before-save-hook)))))
 
 (when typescript-use-tslint
   (defun typescript/init-flycheck-typescript-tslint ()
     (use-package flycheck-typescript-tslint
-    :defer t)))
+      :defer t)))
+
+(defun typescript/init-url ()
```

Comment:
Nope, there is only one layer owning a package, and in this case it's spacemacs-base. There is no reason to add it here, especially as you don't add any configuration.